sentry-android-core/src/main/java/io/sentry/android/core/AndroidTransactionProfiler.java

```diff
@@ -175,14 +185,72 @@ public synchronized void onTransactionStart(@NotNull ITransaction transaction) {
             transactionsCounter);
   }
 
+  @SuppressLint("NewApi")
+  private void onFirstTransactionStarted(
+      final @NotNull ITransaction transaction, final @NotNull File traceFile) {
+
+    measurementsMap.clear();
+    screenFrameRateMeasurements.clear();
+    slowFrameRenderMeasurements.clear();
+    frozenFrameRenderMeasurements.clear();
+
+    frameMetricsCollectorId =
+        frameMetricsCollector.startCollection(
+            new SentryFrameMetricsCollector.FrameMetricsCollectorListener() {
+              final long nanosInSecond = TimeUnit.SECONDS.toNanos(1);
+              final long frozenFrameThresholdNanos = TimeUnit.MILLISECONDS.toNanos(700);
+              float lastRefreshRate = 0;
+
+              @Override
+              public void onFrameMetricCollected(
+                  @NotNull FrameMetrics frameMetrics, float refreshRate) {
+                long frameTimestampRelativeNanos =
+                    SystemClock.elapsedRealtimeNanos() - transactionStartNanos;
+                long durationNanos = frameMetrics.getMetric(FrameMetrics.TOTAL_DURATION);
+                // Most frames take just a few nanoseconds longer than the optimal calculated
+                // duration.
+                // Therefore we subtract one, because otherwise almost all frames would be slow.
+                boolean isSlow = durationNanos > nanosInSecond / (refreshRate - 1);
```

Comment:
I like this approach, we should probably also switch over in the ActivityFramesTracker, because for now we're just using hardcoded 16ms for detecting slow frames